<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>992</width>
    <height>695</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>40</y>
      <width>331</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Qt to STM32 to interface&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>109</x>
      <y>100</y>
      <width>831</width>
      <height>541</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab_1">
     <attribute name="title">
      <string>setup</string>
     </attribute>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>40</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>port</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>90</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>speed</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>130</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>databit</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>170</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>parity</string>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox_port">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>40</y>
        <width>86</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox_speed">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>90</y>
        <width>86</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox_databit">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>130</y>
        <width>86</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox_parity">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>170</y>
        <width>86</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>370</y>
        <width>81</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>port status</string>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>370</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="textVisible">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_scanagain">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>220</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>scanagain</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_open">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>270</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>open</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_close">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>320</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>close</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_send">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>40</y>
        <width>361</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_send">
      <property name="geometry">
       <rect>
        <x>640</x>
        <y>40</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>send</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_receive">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>130</y>
        <width>361</width>
        <height>261</height>
       </rect>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>390</x>
        <y>100</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>receive data</string>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_temp">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>410</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_wet">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>410</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>450</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>온도</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>450</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>습도</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_clear">
      <property name="geometry">
       <rect>
        <x>640</x>
        <y>130</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>CLEAR</string>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox_dht11">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>410</y>
        <width>171</width>
        <height>25</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>1500</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>2000</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>2500</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>3000</string>
       </property>
      </item>
     </widget>
     <widget class="QPushButton" name="pushButton_dht11">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>410</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>send</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>DHT11_Control</string>
     </attribute>
     <widget class="QCustomPlot" name="plot1" native="true">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>100</y>
        <width>291</width>
        <height>311</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_clear_temp">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>430</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>CLEAR</string>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_temp_current">
      <property name="geometry">
       <rect>
        <x>54</x>
        <y>70</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_temp_max">
      <property name="geometry">
       <rect>
        <x>154</x>
        <y>70</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_temp_min">
      <property name="geometry">
       <rect>
        <x>254</x>
        <y>70</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>58</x>
        <y>40</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>CURRENT</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>157</x>
        <y>40</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>MAX</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>259</x>
        <y>40</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>MIN</string>
      </property>
     </widget>
     <widget class="QCustomPlot" name="plot2" native="true">
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>100</y>
        <width>291</width>
        <height>311</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_humid_min">
      <property name="geometry">
       <rect>
        <x>666</x>
        <y>63</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_humid_max">
      <property name="geometry">
       <rect>
        <x>566</x>
        <y>63</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_24">
      <property name="geometry">
       <rect>
        <x>571</x>
        <y>33</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>MAX</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_25">
      <property name="geometry">
       <rect>
        <x>671</x>
        <y>33</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>MIN</string>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_humid_current">
      <property name="geometry">
       <rect>
        <x>466</x>
        <y>63</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_26">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>33</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>CURRENT</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_clear_humid">
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>430</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>CLEAR</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_27">
      <property name="geometry">
       <rect>
        <x>156</x>
        <y>10</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Temp</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_28">
      <property name="geometry">
       <rect>
        <x>568</x>
        <y>10</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Humid</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>STM32Control</string>
     </attribute>
     <widget class="QCheckBox" name="checkBox">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>150</y>
        <width>92</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>led2</string>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar_led">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>150</y>
        <width>118</width>
        <height>23</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>exit</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QLabel" name="label_Alive">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>66</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Alive</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>992</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
